### ISSUE TITLE
[1.4.0rc*] PY3: login using http basic auth is broken

### ISSUE THREAD

**Issue by foosel** [foosel/OctoPrint#3384]
**login using http basic auth is broken**

```
Traceback (most recent call last):
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/tornado/web.py", line 1489, in _execute
    result = self.prepare()
  File "/home/bzed/tools/OctoPrint/src/octoprint/server/util/tornado.py", line 257, in prepare
    self._fallback(self.request, b"")
  File "/home/bzed/tools/OctoPrint/src/octoprint/server/util/tornado.py", line 619, in __call__
    WsgiInputContainer.environ(request, body), start_response)
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1610, in full_dispatch_request
    rv = self.preprocess_request()
  File "/home/bzed/tools/OctoPrint/venv/lib/python3.7/site-packages/flask/app.py", line 1831, in preprocess_request
    rv = func()
  File "/home/bzed/tools/OctoPrint/src/octoprint/server/util/__init__.py", line 55, in loginFromAuthorizationHeaderRequestHandler
    if loginUserFromAuthorizationHeader():
  File "/home/bzed/tools/OctoPrint/src/octoprint/server/util/__init__.py", line 83, in loginUserFromAuthorizationHeader
    user = get_user_for_authorization_header(authorization_header)
  File "/home/bzed/tools/OctoPrint/src/octoprint/server/util/__init__.py", line 240, in get_user_for_authorization_header
    name, password = header.split(':', 1)
TypeError: a bytes-like object is required, not 'str'
```

I *think* the fix is

```
diff --git a/src/octoprint/server/util/__init__.py
b/src/octoprint/server/util/__init__.py
index 3e9a8c124..7b2dac664 100644
--- a/src/octoprint/server/util/__init__.py
+++ b/src/octoprint/server/util/__init__.py
@@ -237,7 +237,7 @@ def get_user_for_authorization_header(header):
        except TypeError:
                return None

-       name, password = header.split(':', 1)
+       name, password = str(header).split(':', 1)
        if not octoprint.server.userManager.enabled:
                return None
diff --git a/src/octoprint/server/util/__init__.py
b/src/octoprint/server/util/__init__.py
index 3e9a8c124..7b2dac664 100644
--- a/src/octoprint/server/util/__init__.py
+++ b/src/octoprint/server/util/__init__.py
@@ -237,7 +237,7 @@ def get_user_for_authorization_header(header):
        except TypeError:
                return None

-       name, password = header.split(':', 1)
+       name, password = str(header).split(':', 1)
        if not octoprint.server.userManager.enabled:
                return None
```

unfortunately something else fscked up the login, so I did not test it properly.

_Originally posted by @bzed in https://github.com/foosel/OctoPrint/issues/3382#issuecomment-564281425_

**Comment by foosel** [comment]
Fixed in 9f85b54

**Comment by foosel** [comment]
1.4.0rc3 has just been released
